packages/lab/src/date-input/DateInput.tsx

```diff
@@ -115,272 +119,291 @@ export interface DateInputProps
    */
   selectionVariant?: "default" | "range";
   /**
-   * Callback fired when the selected daate change.
+   * Callback fired when the selected date change.
    */
   onSelectionChange?: (
     event: SyntheticEvent,
-    selectedDate?: DateValue | { startDate?: DateValue; endDate?: DateValue }
+    selectedDate?: SelectionVariantType
+  ) => void;
+  onChange?: (
+    event: ChangeEvent<HTMLInputElement>,
+    startDateInputValue?: string,
```

Comment:
Given the other change, should input change be typed similarly as well?

Reply:
do you mean with a generic? 
if so, i added the 2 strings separately because that way the first value you get is your single/start date and the next (optional) would be the range, else if i send an object with 2 strings the user would have to destructure it,  but happy to change it

Reply:
Was thinking on single date input, what will be "start" / "end" in that use case